### PR TITLE
Feature/ddns no key

### PR DESCRIPTION
**Pull request by dschaaff**
In its current state the module only allows for turning on ddns updates if a key is provided. There are cases where insecure updates may be desired. For example when running isn-dhcp-server along with Microsoft DNS servers. This pull request creates 1 new variable, ddns_updates. If this variable is set then the template will add the ddns configs without the key.

### CODE REVIEW

File: templates/dhcpd.conf.erb [hunk 1]
```diff
@@ -17,7 +17,7 @@ max-lease-time <%= @max_lease_time %>;
 authoritative;
 <% end -%>
 
-<% if has_variable?( 'dnsupdatekey' ) and @dnsupdatekey -%>
+<% if has_variable?( 'dnsupdatekey' ) and @dnsupdatekey or has_variable?( 'ddns_updates' ) and @ddns_updates -%>
```
Comment (domcleal) [inline templates/dhcpd.conf.erb]: 1. Use brackets here to contain the two parts of the expression, it's dependent on precedence of `and`/`or` right now.
2. Prefer `&&` and `||` in Ruby/ERB expressions, not `and`/`or`.

File: templates/dhcpd.conf.erb [hunk 2]
```diff
@@ -91,4 +95,4 @@ include "<%= include %>";
 <% end -%>
 <% elsif @includes && !@includes.strip.empty? -%>
 include "<%= @includes %>";
-<% end -%>
+<% end -%>
```
Comment (domcleal) [inline templates/dhcpd.conf.erb]: Nitpick, your editor's removed the EOL, please keep it.

**Comment by domcleal** [comment]
Thanks for the PR.  Could you check the test failures and squash your commits into one?  If you're able to add a test as well, that'd be much appreciated.

**Comment by dschaaff** [comment]
Thanks for the feedback. I've pushed the requested changes with the exception of the tests.

**Comment by ekohl** [comment]
@dschaaff thanks for your contribution, even if it took us a long time to merge it.

**Comment by dschaaff** [comment]
Thanks for the merge!

**Pull request closed without merging**